### PR TITLE
Implement listener

### PR DESCRIPTION
**Pull request by oggg**
Implement cautionary alert listener

### CODE REVIEW

File: CautionaryAlertsListener.Tests/UseCase/PersonUpdatedUseCaseTests.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            var mmhId = _message.Id;
+            _mockGateway.Setup(x => x.GetEntitiesByMMHAndPropertyReferenceAsync(It.IsAny<Guid>().ToString(), null))
+                .ReturnsAsync(new List<PropertyAlertNew>());
+
+            Func<Task> func = async () => await _sut.ProcessMessageAsync(_message).ConfigureAwait(false);
+            func.Should().NotThrow();
+            _mockGateway.Verify(x => x.UpdateEntityAsync(It.IsAny<PropertyAlertNew>()), Times.Never);
+        }
+
+        // Comment this out as the Expandoobject data is serialized in SQS message and not when the use case is tested
```
Comment (humulla) [inline CautionaryAlertsListener.Tests/UseCase/PersonUpdatedUseCaseTests.cs]: Is this comment still required?

File: CautionaryAlertsListener/serverless.yml [hunk 2]
```diff
@@ -8,25 +8,20 @@ provider:
...
-  BaseListener:
+  CautionaryAlertsListener:
     name: ${self:service}-${self:provider.stage}
-    handler: BaseListener::BaseListener.SqsFunction::FunctionHandler
+    handler: CautionaryAlertsListener::CautionaryAlertsListener.CautionaryAlertsListener::FunctionHandler
     role: lambdaExecutionRole
     environment:
-      ### Re-use the same parameter as used by web api applications
-      ENVIRONMENT:  ${ssm:/housing-tl/${self:provider.stage}/aspnetcore-environment}
+      CONNECTION_STRING: Host=${ssm:/uh-api/${self:provider.stage}/postgres-hostname};Port=${ssm:/uh-api/${self:provider.stage}/postgres-port};Database=uh_mirror;Username=${ssm:/uh-api/${self:provider.stage}/postgres-username};Password=${ssm:/uh-api/${self:provider.stage}/postgres-password}
```
Comment (humulla) [inline CautionaryAlertsListener/serverless.yml:22]: You need to add the other environment variables as well, such as tenureApiToken and tenureApiUrl

File: terraform/development/main.tf [hunk 3]
```diff
@@ -34,84 +42,90 @@ terraform {
...
-
-### This creates an AWS parameter with arn of the queue that will then be used within the Serverless.yml
-### to specify the queue that will trigger the lambda function.
-# 
-# resource "aws_ssm_parameter" "THIS_DOMAIN_sqs_queue_arn" {
-#   name  = "/sqs-queue/development/THIS_DOMAIN/arn"
-#   type  = "String"
-#   value = aws_sqs_queue.THIS_DOMAIN_queue.arn
-# }
+module "housing_search_listener_cw_dashboard" {
```
Comment (humulla) [inline terraform/development/main.tf]: This should be renamed to `cautionary_alerts_listener_cw_dashboard`

File: terraform/production/main.tf [hunk 4]
```diff
@@ -33,84 +42,90 @@ terraform {
...
+  raw_message_delivery = true
+}
+
+resource "aws_ssm_parameter" "cautionary_alerts_listener_sqs_queue_arn" {
+  name  = "/sqs-queue/production/cautionary_alerts_listener_queue/arn"
+  type  = "String"
+  value = aws_sqs_queue.cautionary_alerts_listener_queue.arn
+}
+
+module "housing_search_listener_cw_dashboard" {
```
Comment (humulla) [inline terraform/production/main.tf]: This should be renamed to `cautionary_alerts_listener_cw_dashboard`

File: terraform/staging/main.tf [hunk 5]
```diff
@@ -33,85 +42,90 @@ terraform {
...
+resource "aws_sqs_queue" "cautionary_alerts_dead_letter_queue" {
+  name                        = "cautionaryalertsdeadletterqueue.fifo"
+  fifo_queue                  = true
+  content_based_deduplication = true
+  kms_master_key_id           = "alias/housing-staging-cmk"
+  kms_data_key_reuse_period_seconds = 300
+}
+
+resource "aws_sqs_queue" "cautionary_alerts_listener_queue" {
+  name                              = "housingsearchqueue.fifo"
```
Comment (humulla) [inline terraform/staging/main.tf]: The name should be `cautionaryalertsqueue.fifo`

File: terraform/staging/main.tf [hunk 6]
```diff
@@ -33,85 +42,90 @@ terraform {
...
+  raw_message_delivery = true
+}
+
+resource "aws_ssm_parameter" "cautionary_alerts_listener_sqs_queue_arn" {
+  name  = "/sqs-queue/staging/cautionary_alerts_listener_queue/arn"
+  type  = "String"
+  value = aws_sqs_queue.cautionary_alerts_listener_queue.arn
+}
 
+module "housing_search_listener_cw_dashboard" {
```
Comment (humulla) [inline terraform/staging/main.tf]: This should be renamed to `cautionary_alerts_listener_cw_dashboard`

File: terraform/production/main.tf [hunk 7]
```diff
@@ -33,84 +42,90 @@ terraform {
...
+resource "aws_sqs_queue" "cautionary_alerts_dead_letter_queue" {
+  name                        = "cautionaryalertsdeadletterqueue.fifo"
+  fifo_queue                  = true
+  content_based_deduplication = true
+  kms_master_key_id           = "alias/housing-production-cmk"
+  kms_data_key_reuse_period_seconds = 300
+}
+
+resource "aws_sqs_queue" "cautionary_alerts_listener_queue" {
+  name                              = "housingsearchqueue.fifo"
```
Comment (humulla) [inline terraform/production/main.tf]: The name should be cautionaryalertsqueue.fifo

File: terraform/development/main.tf [hunk 8]
```diff
@@ -34,84 +42,90 @@ terraform {
...
+resource "aws_sqs_queue" "cautionary_alerts_dead_letter_queue" {
+  name                        = "cautionaryalertsdeadletterqueue.fifo"
+  fifo_queue                  = true
+  content_based_deduplication = true
+  kms_master_key_id           = "alias/housing-development-cmk"
+  kms_data_key_reuse_period_seconds = 300
+}
+
+resource "aws_sqs_queue" "cautionary_alerts_listener_queue" {
+  name                              = "housingsearchqueue.fifo"
```
Comment (humulla) [inline terraform/development/main.tf]: The name should be` cautionaryalertsqueue.fifo`

File: CautionaryAlertsListener.Tests/FixtureConstants.cs [hunk 9]
```diff
@@ -0,0 +1,12 @@
+namespace CautionaryAlertsListener.Tests
+{
+    public static class FixtureConstants
```
Comment (humulla) [inline CautionaryAlertsListener.Tests/FixtureConstants.cs:3]: If this is used only as part of the E2E tests then can we move this to be in the E2E folder

File: CautionaryAlertsListener.Tests/EventDataFixture.cs [hunk 10]
```diff
@@ -0,0 +1,25 @@
+using Hackney.Core.Sns;
+using System.Dynamic;
+
+namespace CautionaryAlertsListener.Tests
+{
+    public static class EventDataFixture
```
Comment (humulla) [inline CautionaryAlertsListener.Tests/EventDataFixture.cs]: If this is used only as part of the E2E tests then can we move this to be in the E2E folder

File: CautionaryAlertsListener.Tests/CreateCautionaryAlertFixture.cs [hunk 11]
```diff
@@ -0,0 +1,41 @@
+using AutoFixture;
+using Hackney.Shared.CautionaryAlerts.Boundary.Request;
+using Hackney.Shared.CautionaryAlerts.Infrastructure;
+using System;
+
+namespace CautionaryAlertsListener.Tests
+{
+    public static class CreateCautionaryAlertFixture
+    {
+        public static CreateCautionaryAlert GenerateValidCreateCautionaryAlertFixture(string defaultString, Fixture fixture, Guid? mmhId = null, string propertyReference = null)
```
Comment (humulla) [inline CautionaryAlertsListener.Tests/CreateCautionaryAlertFixture.cs:10]: Is this being used anywhere?

**Review by humulla — COMMENTED** [review]
The pipeline is failing as well.

File: CautionaryAlertsListener.Tests/FixtureConstants.cs (hunk 9 above)
Comment (oggg) [reply CautionaryAlertsListener.Tests/FixtureConstants.cs:3]: It is used by other tests too (use cases, API fixtures, etc)

File: CautionaryAlertsListener.Tests/CreateCautionaryAlertFixture.cs (hunk 11 above)
Comment (oggg) [reply CautionaryAlertsListener.Tests/CreateCautionaryAlertFixture.cs:10]: image

**Review by humulla — APPROVED** [review]

**Pull request merged**